src/app/providers/vehicle-details-api/vehicle-details-api.service.ts

```diff
@@ -21,22 +31,33 @@ export class VehicleDetailsApiService {
   vehicleIdentifier: string;
   vehicleDetailsResponse: VehicleDetails;
 
-  getVehicleByIdentifier(vehicleRegistration: string) {
+  getVehicleByIdentifier(vehicleRegistration: string): Observable<MotDataWithStatus> {
     if (vehicleRegistration === this.vehicleIdentifier && this.vehicleDetailsResponse !== undefined) {
-      return of(this.vehicleDetailsResponse);
+      return of({ status: 'Already Saved', data: this.vehicleDetailsResponse });
     }
 
     const headers = new HttpHeaders().set('x-api-key', this.urlProvider.getTaxMotApiKey());
     const params = new HttpParams().set('identifier', vehicleRegistration);
 
     return this.http.get(
-      this.urlProvider.getTaxMotUrl(), { headers, params },
+      this.urlProvider.getTaxMotUrl(), { observe: 'response', headers, params },
     ).pipe(
-      tap((response: VehicleDetails) => {
-        this.vehicleIdentifier = vehicleRegistration;
-        this.vehicleDetailsResponse = response;
+      tap((response: HttpResponse<VehicleDetails>) => {
+        if (response.status === HttpStatusCodes.OK) {
+          this.vehicleIdentifier = response.body.registration;
+          this.vehicleDetailsResponse = response.body;
+        }
+      }),
+      map((value):MotDataWithStatus => {
+        return {
+          status: value.status.toString(),
+          data: value.body,
+        };
       }),
       timeout(this.appConfig.getAppConfig().requestTimeout),
+      catchError((err) => {
+        return of({ status: err.status, data: null });
+      }),
     );
```

Comment:
There could be a number of performance improvements implemented here, if vehicle is not found then why would we re-query the endpoint - same for if there is no content etc.

Reply:
Catch error is being reported here which presumably means the catch error in the effect is now no longer working